### PR TITLE
Variable registry update

### PR DESCRIPTION
**Pull request by 0ctagon**
This PR adds a lot of functionality to the variable registry.

Link to updated doc: https://plothist.readthedocs.io/en/registry-update/basics/variable_registry.html

### CODE REVIEW

**Review by cyrraz — CHANGES_REQUESTED** [review]

File: plothist/variable_registry.py [hunk 1]
```diff
@@ -103,66 +155,101 @@ def get_variable_from_registry(variable, path="./variable_registry.yaml"):
...
     with open(path, "r") as f:
         variable_registry = yaml.safe_load(f)
-    variable_registry[variable_key]["range"] = [x_min, x_max]
+    print(variable_registry)
+    if variable_keys is None:
+        variable_keys = list(variable_registry.keys())
+
+    for variable_key in variable_keys:
+        for parameter in parameters:
+            try:
```
[the remark this answers is not in the document]
Comment (cyrraz) [reply plothist/variable_registry.py:231]: Or better:

```
if variable_registry[variable_key].pop(parameter, None) is None:
    warnings.warn(f"{parameter} not present in the registry for {variable_key}, skipping.")
```

[the remark this answers is not in the document]
Comment (0ctagon) [reply plothist/variable_registry.py:231]: I thonk copilot learnt to much ROOT...

File: plothist/variable_registry.py [hunk 2]
```diff
@@ -103,91 +156,142 @@ def get_variable_from_registry(variable, path="./variable_registry.yaml"):
...
+                    f"{parameter} parameter not present in the registry {path} for {variable_key}, skipping."
+                )
 
+    _save_variable_registry(variable_registry, path=path)
 
-def update_variable_registry_ranges(data, variables, path="./variable_registry.yaml"):
+
+def update_variable_registry_ranges(
+    data,
+    variable_keys,
```
Comment (cyrraz) [inline plothist/variable_registry.py:244]: `=None`

File: plothist/variable_registry.py [hunk 3]
```diff
@@ -103,91 +156,142 @@ def get_variable_from_registry(variable, path="./variable_registry.yaml"):
...
         variable = get_variable_from_registry(variable_key, path=path)
-        axis = create_axis(data[variable_key], variable["bins"], variable["range"])
-        if isinstance(axis, bh.axis.Regular):
-            update_variable_registry(
-                variable_key, float(axis.edges[0]), float(axis.edges[-1]), path=path
-            )
-        else:
-            raise NotImplemented(
-                f"Only regular binning allowed in registry. {type(axis)}"
+        try:
```
Comment (cyrraz) [inline plothist/variable_registry.py:280]: Why not `if "bins" in variable.keys() and "range" in variable.keys():`

File: plothist/variable_registry.py [hunk 4]
```diff
@@ -103,91 +156,142 @@ def get_variable_from_registry(variable, path="./variable_registry.yaml"):
...
+    _check_if_variable_registry_exists(path)
 
     with open(path, "r") as f:
         variable_registry = yaml.safe_load(f)
         return variable_registry[variable]
 
 
 def update_variable_registry(
-    variable_key, x_min, x_max, path="./variable_registry.yaml"
+    dictionary, variable_keys, path="./variable_registry.yaml", overwrite=False
```
Comment (cyrraz) [inline plothist/variable_registry.py:167]: `variable_keys=None`

**Review by cyrraz — CHANGES_REQUESTED** [review]

**Review by cyrraz — APPROVED** [review]

**Pull request merged**